src/scipp/core/operations.py

```diff
@@ -358,16 +359,23 @@ def _merge_data_group(lhs: DataGroup, rhs: DataGroup) -> DataGroup:
     return res
 
 
-def label_based_to_positional_index(
+def label_based_index_to_positional_index(
     sizes: Dict[str, int], coord: Variable, *values: Variable
 ) -> ScippIndex:
     """Returns the positional index equivalent to label based indexing
-    on coord with limits values."""
+    the coord with values."""
+    first, *rest = values
+    first = scalar(float('-inf'), unit=coord.unit) if first is None else first
+    rest = (
+        scalar(float('inf'), unit=coord.unit) if second is None else second
+        for second in rest
+    )
```

Comment:
Does not work with `int`? I think we may need to avoid this. Can you pass the limits as optional to deal with this (can't remember how it is implemented in the label-based indexing)?

Reply:
What does not work with `int`? Do you mean that if the coord has dtype `int` the comparison will not work, or did you mean something different?

The best would be if we could just pass the `None` values directly to the c++ implementation. Unfortunately `get_slice_params` does not handle that case, it's done here: https://github.com/scipp/scipp/blob/f99f9db457d5fd665fe34f8b100825ee967cb60e/lib/python/bind_slice_methods.h#L88.

Reply:
> What does not work with int? Do you mean that if the coord has dtype int the comparison will not work, or did you mean something different?

Yes, the dtype will be wrong.

> The best would be if we could just pass the None values directly to the c++ implementation. Unfortunately get_slice_params does not handle that case, it's done here: 

Why can't we use that function here (or at least the part inside the `try` block)? It looks like it is exactly what we need, and we'd even get consistent exception messages.

Reply:
Yeah we can probably use that implementation instead. I'll try it out.

I think it works with integer dtype though, look at the test just added for that, or did you mean something different?

Reply:
The tests does not prove much. If an open bound causes conversion of the other int bound to float you may lose precision, which is bad.